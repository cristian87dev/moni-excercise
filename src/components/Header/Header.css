.header {
  display: flex;
  flex-direction: column;

  padding: 0.25rem 1rem;
  box-shadow: 0 2px 5px rgba(0,0,0,0.15);

  @media (min-width: 768px) {
    flex-direction: row;
  }
}

.title {
  margin: 0;
  padding: 0.25rem 0;
  text-align: center;

  @media (min-width: 768px) {
    padding: 0;
    text-align: center;
  }
}
