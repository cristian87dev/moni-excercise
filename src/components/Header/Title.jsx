import React from 'react';
import { Link } from 'react-router-dom';

import './Header.css';
import logo from 'assets/images/moni-logo.svg';

const HeaderTitle = () => (
  <h1 className="title">
    <Link to="/">
      <img src={logo} alt="Moni Logo" />
    </Link>
  </h1>
);

export default HeaderTitle;
