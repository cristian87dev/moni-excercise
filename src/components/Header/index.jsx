import React from 'react';

import Nav from 'components/Nav';
import HeaderTitle from 'components/Header/Title';

import './Header.css';

const Header = () => (
  <header className="header">
    <HeaderTitle />
    <Nav />
  </header>
);

export default Header;
