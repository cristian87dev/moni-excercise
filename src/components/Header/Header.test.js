import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom'; // To prevent Link dependency error.
import renderer from 'react-test-renderer';
import Header from './index';

it('Header renders without crashing', () => {
  const component = renderer.create(
    <Router>
      <Header />
    </Router>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
