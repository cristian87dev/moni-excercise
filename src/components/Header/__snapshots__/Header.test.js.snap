// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
<header
  className="header"
>
  <h1
    className="title"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Moni Logo"
        src="moni-logo.svg"
      />
    </a>
  </h1>
  <nav
    className="nav-menu"
  >
    <ul>
      <li
        className="nav-item"
      >
        <a
          className="link-item"
          href="/pedi-tu-prestamo"
          onClick={[Function]}
        >
          Ped√≠ tu Prestamo
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="link-item"
          href="/user-admin"
          onClick={[Function]}
        >
          User Admin
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
